refl1d/probe/data_loaders/load4.py

```diff
@@ -87,26 +87,46 @@ def get_key(orso_name, refl1d_name, refl1d_resolution_name):
                 (i for i, c in enumerate(columns) if getattr(c, "physical_quantity", None) == orso_name),
                 None,
             )
+            resolution_index = None
```

Comment:
get_key() does not need to be defined in the for loop. It picks up the local scope regardless of where things are defined, so long as it is defined before it is used.